include/ruby/re.h

```diff
@@ -105,25 +106,28 @@ long rb_reg_adjust_startpos(VALUE re, VALUE str, long pos, int dir);
 VALUE rb_reg_quote(VALUE str);
 
 /**
- * Exercises  various  checks  and  preprocesses  so  that  the  given  regular
- * expression can be applied to the given string.  The preprocess here includes
- * (but not limited to) for instance encoding conversion.
+ * Runs a regular expression match using function `match`. Performs preparation,
+ * error handling, and memory cleanup.
  *
  * @param[in]  re                  Target regular expression.
  * @param[in]  str                 What `re` is about to run on.
+ * @param[in]  match               The function to run to match `str` against `re`.
+ * @param[in]  args                Pointer to arguments to pass into `match`.
+ * @param[out] regs                Registers on a successful match.
  * @exception  rb_eArgError        `re` does not fit for `str`.
  * @exception  rb_eEncCompatError  `re` and `str` are incompatible.
  * @exception  rb_eRegexpError     `re` is malformed.
- * @return     A preprocessesed pattern buffer ready to be applied to `str`.
- * @note       The return value is manages by our GC.  Don't free.
+ * @return     Match position on a successful match, `ONIG_MISMATCH` otherwise.
  *
  * @internal
  *
- * The  return  type,  `regex_t  *`, is  defined  in  `<ruby/onigmo.h>`,  _and_
+ * The type `regex_t  *` is  defined  in  `<ruby/onigmo.h>`,  _and_
  * _conflicts_ with POSIX's  `<regex.h>`.  We can no longer  save the situation
  * at this point.  Just don't mix the two.
  */
-regex_t *rb_reg_prepare_re(VALUE re, VALUE str);
```

Comment:
I understand that you changed strscan to stop using this, but this is still a C API exposed by this public header. Are you sure existing gems don't use this?

Reply:
Well, strscan 3.0.7 has not been released yet, so I cannot even run railsbench with Ruby master anymore. Let me re-expose the API for now.

Reply:
done https://github.com/ruby/ruby/commit/97219721752e7cc1eda51131d6a3cd753d37276c

Reply:
I looked through gem-codesearch and found that the only gems that use it are abandoned ([match_at](https://rubygems.org/gems/match_at) and [mmapscanner](https://rubygems.org/gems/mmapscanner)).

I forgot that previous versions of strscan installed in a Gemfile still needs this function. Thanks for fixing it!